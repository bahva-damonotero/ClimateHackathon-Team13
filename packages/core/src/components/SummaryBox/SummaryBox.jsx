import React from 'react';
import PropTypes from 'prop-types';

export default function SummaryBox({ children, heading }) {
  return (
    <div className="usa-summary-box" role="complementary">
      <div className="usa-summary-box__body">
        <h3 className="usa-summary-box__heading">{heading}</h3>

        <div className="usa-summary-box__text">
          <ul className="usa-list">{children}</ul>
        </div>
      </div>
    </div>
  );
}

SummaryBox.propTypes = {
  /** `<li>` node(s) */
  children: PropTypes.node.isRequired,
  /** Heading content for the SummaryBox */
  heading: PropTypes.node.isRequired
};
