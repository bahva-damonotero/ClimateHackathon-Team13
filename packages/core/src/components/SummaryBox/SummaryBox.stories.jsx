import React from 'react';

import SummaryBox from '.';

export default {
  title: 'SummaryBox',
  component: SummaryBox,
  argTypes: {
    children: { control: { type: null } }
  }
};

const Template = (args) => <SummaryBox {...args} />;

export const def = Template.bind({});
def.storyName = 'Default';
def.args = {
  children: (
    <>
      <li>
        If you are under a winter storm warning,{' '}
        <a className="usa-summary-box__link" href="#usa-anchor-find-shelter">
          find shelter
        </a>{' '}
        right away.
      </li>

      <li>
        Sign up for{' '}
        <a className="usa-summary-box__link" href="#usa-anchor-warning-system">
          your communityâ€™s warning system
        </a>
        .
      </li>

      <li>
        Learn the signs of, and basic treatments for,{' '}
        <a className="usa-summary-box__link" href="#usa-anchor-frostbite">
          frostbite
        </a>{' '}
        and{' '}
        <a className="usa-summary-box__link" href="#usa-anchor-hypothermia">
          hypothermia
        </a>
        .
      </li>

      <li>
        Gather emergency supplies for your{' '}
        <a className="usa-summary-box__link" href="#usa-anchor-home">
          home
        </a>{' '}
        and your{' '}
        <a className="usa-summary-box__link" href="#usa-anchor-car">
          car
        </a>
        .
      </li>
    </>
  ),
  heading: 'Key information'
};
