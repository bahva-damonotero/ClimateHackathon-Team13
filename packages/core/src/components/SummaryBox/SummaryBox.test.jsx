import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import SummaryBox from '.';

describe('SummaryBox', () => {
  const heading = 'Key information';

  const jsx = (
    <SummaryBox heading={heading}>
      <li>
        If you are under a winter storm warning,{' '}
        <a className="usa-summary-box__link" href="#usa-anchor-find-shelter">
          find shelter
        </a>{' '}
        right away.
      </li>

      <li>
        Sign up for{' '}
        <a className="usa-summary-box__link" href="#usa-anchor-warning-system">
          your communityâ€™s warning system
        </a>
        .
      </li>

      <li>
        Learn the signs of, and basic treatments for,{' '}
        <a className="usa-summary-box__link" href="#usa-anchor-frostbite">
          frostbite
        </a>{' '}
        and{' '}
        <a className="usa-summary-box__link" href="#usa-anchor-hypothermia">
          hypothermia
        </a>
        .
      </li>

      <li>
        Gather emergency supplies for your{' '}
        <a className="usa-summary-box__link" href="#usa-anchor-home">
          home
        </a>{' '}
        and your{' '}
        <a className="usa-summary-box__link" href="#usa-anchor-car">
          car
        </a>
        .
      </li>
    </SummaryBox>
  );

  it('should render correctly', () => {
    const { container } = render(jsx);

    expect(container.querySelector('.usa-summary-box')).toBeTruthy();
    expect(container.querySelector('.usa-summary-box__heading').innerHTML).toBe(
      heading
    );
  });

  it('should match snapshot', () => {
    const tree = renderer.create(jsx).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
