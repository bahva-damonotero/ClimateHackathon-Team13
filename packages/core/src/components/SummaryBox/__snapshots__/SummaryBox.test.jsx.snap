// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryBox should match snapshot 1`] = `
<div
  className="usa-summary-box"
  role="complementary"
>
  <div
    className="usa-summary-box__body"
  >
    <h3
      className="usa-summary-box__heading"
    >
      Key information
    </h3>
    <div
      className="usa-summary-box__text"
    >
      <ul
        className="usa-list"
      >
        <li>
          If you are under a winter storm warning,
           
          <a
            className="usa-summary-box__link"
            href="#usa-anchor-find-shelter"
          >
            find shelter
          </a>
           
          right away.
        </li>
        <li>
          Sign up for
           
          <a
            className="usa-summary-box__link"
            href="#usa-anchor-warning-system"
          >
            your communityâ€™s warning system
          </a>
          .
        </li>
        <li>
          Learn the signs of, and basic treatments for,
           
          <a
            className="usa-summary-box__link"
            href="#usa-anchor-frostbite"
          >
            frostbite
          </a>
           
          and
           
          <a
            className="usa-summary-box__link"
            href="#usa-anchor-hypothermia"
          >
            hypothermia
          </a>
          .
        </li>
        <li>
          Gather emergency supplies for your
           
          <a
            className="usa-summary-box__link"
            href="#usa-anchor-home"
          >
            home
          </a>
           
          and your
           
          <a
            className="usa-summary-box__link"
            href="#usa-anchor-car"
          >
            car
          </a>
          .
        </li>
      </ul>
    </div>
  </div>
</div>
`;
